library/gd_mf_weights.cc

```diff
@@ -58,8 +58,23 @@ int main(int argc, char *argv[])
   size_t rank = atoi(location);
 
   // global model params
-  unsigned char left_ns = model->pairs[0][0];
-  unsigned char right_ns = model->pairs[0][1];
+  std::vector<unsigned char> first_pair;
```

Comment:
nit: make this a pointer to avoid the copy?

Reply:
This code is a rarely used example binary, going to leave it as is.